dashboard/app/controllers/levels_controller.rb

```diff
@@ -350,4 +353,22 @@ def set_solution_image_url(level)
     )
     @level.properties['solution_image_url'] = level_source_image.s3_url if level_source_image
   end
+
+  # Gathers data on top pain points for level builders by logging error details
+  # to Firehose / Redshift.
+  def log_save_error(level)
+    FirehoseClient.instance.put_record(
+      study: 'level-save-error',
+      # Make it easy to count most frequent field name in which errors occur.
+      event: level.errors.keys.first,
+      # Level ids are different on levelbuilder, so use the level name. This can
+      # be joined against to determine the level type or other level properties.
```

Comment:
joined against what?

Reply:
I meant that I'd join against that column. but perhaps I should say that id join "on" that column, against the levels table. I will update.